qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -367,3 +376,270 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
         return common.generate_scheduling(
             instruction_durations, scheduling_method, timing_constraints, inst_map, target
         )
+
+
+class DefaultLayoutPassManager(PassManagerStagePlugin):
+    """Plugin class for default layout stage."""
+
+    def pass_manager(self, pass_manager_config, optimization_level=None) -> PassManager:
+        _given_layout = SetLayout(pass_manager_config.initial_layout)
+
+        def _choose_layout_condition(property_set):
+            return not property_set["layout"]
+
+        def _layout_not_perfect(property_set):
+            """Return ``True`` if the first attempt at layout has been checked and found to be
+            imperfect.  In this case, perfection means "does not require any swap routing"."""
+            return property_set["is_swap_mapped"] is not None and not property_set["is_swap_mapped"]
+
+        def _vf2_match_not_found(property_set):
+            # If a layout hasn't been set by the time we run vf2 layout we need to
+            # run layout
+            if property_set["layout"] is None:
+                return True
+            # if VF2 layout stopped for any reason other than solution found we need
+            # to run layout since VF2 didn't converge.
+            if (
+                property_set["VF2Layout_stop_reason"] is not None
+                and property_set["VF2Layout_stop_reason"] is not VF2LayoutStopReason.SOLUTION_FOUND
+            ):
+                return True
+            return False
```

Comment:
This is totally pre-existing and doesn't need changing, but
```python
if <boolean_expression>:
  return True
return False
```
is a super unnecessary construction haha

Reply:
Lol, I can collapse it. I'm not sure what I was thinking when I added this (since I assume I wrote this originally as it's vf2layout related).

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10622/commits/359f076e32879f5bc706ab9eb33ad98233a6e200